integration_tests/test_dependency_manager.py

```diff
@@ -0,0 +1,66 @@
+import os
+import subprocess
+import shutil
+import pytest
+from integration_tests.base_test import SAMPLES_DIR, CleanRepoMixin
+
+
+class TestDependencyManager(CleanRepoMixin):
```

Comment:
Shouldn't we have test cases here for updating `pyproject.toml` and `setup.py` as well? In those cases we could just add some simple content to those temporary files within individual test cases.